vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -46,31 +46,28 @@ def validate_multi(multi, info: ValidationInfo):
 
 
 def _add_select_all_option(
-    full_options: OptionsType, component_id: str, value: Optional[Union[SingleValueType, MultiValueType]]
-) -> OptionsType:
+    options: OptionsType, component_id: str, value: Optional[Union[SingleValueType, MultiValueType]]
+) -> list[OptionsDictType]:
     """Adds a 'Select All' option to the list of options."""
-    checklist_value = (
-        [ALL_OPTION] if value is None or (isinstance(value, list) and len(value) == len(full_options)) else []
-    )
-    full_options = cast(list[OptionsDictType], full_options)
-    all_options = {
+    checklist_value = True if value is None or (isinstance(value, list) and len(value) == len(options)) else False
+
+    all_option = {
         "label": html.Div(
             [
-                dcc.Checklist(
-                    options=[{"label": "", "value": ALL_OPTION}],
-                    value=checklist_value,
+                dbc.Checkbox(
                     id=f"{component_id}_checklist_all",
```

Comment:
```suggestion
                    id=f"{component_id}_checkbox_all",
```
or
```suggestion
                    id=f"{component_id}_select_all",
```

Or something like that. Similarly for `className` - let's just try and keep things tidy now we changed this component type.